flink-examples/flink-examples-streaming/src/main/java/org/apache/flink/streaming/examples/windowing/WindowWordCount.java

```diff
@@ -118,15 +129,33 @@ public static void main(String[] args) throws Exception {
                         // Using a keyBy allows performing aggregations and other
                         // stateful transformations over data on a per-key basis.
                         // This is similar to a GROUP BY clause in a SQL query.
-                        .keyBy(value -> value.f0)
-                        // create windows of windowSize records slided every slideSize records
-                        .countWindow(windowSize, slideSize)
-                        // For each key, we perform a simple sum of the "1" field, the count.
-                        // If the input data set is bounded, sum will output a final count for
-                        // each word. If it is unbounded, it will continuously output updates
-                        // each time it sees a new instance of each word in the stream.
-                        .sum(1)
-                        .name("counter");
+                        .keyBy(value -> value.f0);
+
+        DataStream<Tuple2<String, Integer>> counts;
+        if (params.isAsyncState()) {
+            counts =
+                    keyedStream
+                            .enableAsyncState()
+                            // create windows of windowSize records slided every slideSize records
+                            .asyncCountWindow(windowSize, slideSize)
```

Comment:
Ah.... we won't provide new API for user. Only `countWindow` combined with `enableAsyncState` is enough.

Reply:
Resolved, I have removed this API. countWindow() + enableAsyncState() will trigger DataStream using AsyncWindowOperator.